src/action/action_built.cpp

```diff
@@ -331,17 +321,12 @@ void COrder_Built::AiUnitKilled(CUnit &unit)
 }
 
 
-static const CConstructionFrame *FindCFramePercent(const CConstructionFrame &cframe, int percent)
+static std::size_t FindCFramePercent(const std::vector<CConstructionFrame> &cframes, int percent)
 {
-	const CConstructionFrame *prev = &cframe;
+	const auto it =
+		ranges::find_if(cframes, [&](const auto &frame) { return percent < frame.Percent; });
```

Comment:
Is it worth doing binary_search here?
And is it even guaranteed/checked that that Percent-values are ordered correctly?

Reply:
I tested on wargus and it seems fine...
I will recheck.

Reply:
Well spot, I indeed have bad offset.
I also add pre-require check in https://github.com/Wargus/stratagus/pull/638.